src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -773,6 +791,34 @@ public CopyToAsyncStream(DeflateStream deflateStream, Stream destination, int bu
                 }
             }
 
+            public void CopyFromSourceToDestination()
+            {
+                try
+                {
+                    // Flush any existing data in the inflater to the destination stream.
+                    while (true)
+                    {
+                        int bytesRead = _deflateStream._inflater.Inflate(_arrayPoolBuffer, 0, _arrayPoolBuffer.Length);
+                        if (bytesRead > 0)
+                        {
+                            _destination.Write(new ReadOnlySpan<byte>(_arrayPoolBuffer, 0, bytesRead));
```

Comment:
I'd suggest sticking with using the array-based overload.  In the async case, there are significant advantages to using the new Memory-based overload, e.g. ability to avoid pinning, ability to avoid allocating a return task, etc.  In the sync case, though, if we already have an array, there's little benefit to using the span-based overload, and if the destination doesn't yet optimize for it, it could actually be worse.  So, I think this should just be:
```C#
_destination.Write(_arrayPoolBuffer, 0, bytesRead);
```

Reply:
Ah, of course! Will fix.